var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -139,10 +142,20 @@ def cmake_args(self):
         else:
             options.append('-DGMX_HWLOC:BOOL=OFF')
 
-        if '+cuda' in self.spec or '+opencl' in self.spec:
-            options.append('-DGMX_GPU:BOOL=ON')
+        if version >= Version('2021'):
+            if '+cuda' in self.spec:
+                options.append('-DGMX_GPU:STRING=CUDA')
+            if '+opencl' in self.spec:
+                options.append('-DGMX_GPU:STRING=OpenCL')
+            if '+sycl' in self.spec:
+                options.append('-DGMX_GPU:STRING=SYCL')
+            else:
```

Comment:
that does not work es expected; this 'else:' will be true for all cases except +sycl, i.e +cuda will result in both options.append('-DGMX_GPU:STRING=CUDA') **and** options.append('-DGMX_GPU:STRING=OFF').
It probably should be ``elif '+opencl' in self.spec:`` ... ``elif '+sycl' in self.spec:`` ... `else:` as I expect that the three options are mutually exclusive; otherwise the ``else:`` has to be changed into "if not any of cuda|opencl|sycl" ...

Reply:
used the if-elif-else format. I agree that they should be mutually exclusive.